src/Tokenizer.ts

```diff
@@ -1,45 +1,93 @@
+import * as fs from "fs";
+import * as path from "path";
+import { ParserError } from "./errors/ParserError";
 
-class Tokenizer {
-  program: String = "";
-  literals: Array<String> = ["start", "Events:", "every", "and", "on", "all day", "from", "at", "with", "Done", "End"];
-  tokens: Array<Sting> = new Array<String>;
-  currentToken: int;
-  tokenizer: Tokenizer;
-
-  // TODO space happy tokens
-
-  getNext(): String {
-    let token: String = "";
-    if (this.moreTokens()) {
-      token = this.tokens[this.currentToken];
-      this.currentToken++;
-    } else {
-      token = "NULLTOKEN";
+export default class Tokenizer {
+  program: string;
+  literals: Array<string> = [
+    "Start",
+    "Events:",
+    "every",
+    "and",
+    "on",
+    "all day",
+    "from",
+    "at",
+    "with",
+    "Done",
+    "End"
+  ];
+  tokens: string[] = [];
+  currentTokenIdx: number = 0;
+  line: number = 1;
+  column: number = 0;
+
+  constructor(fileName: string) {
+    try {
+      this.program = fs
+        .readFileSync(path.join(__dirname, "../", fileName))
+        .toString("utf-8");
+    } catch (err) {
+      throw new ParserError(`Unable to load source: ${fileName}`);
     }
-    return token;
+    this.tokenize();
   }
 
-  checkToken(regex): boolean {
-    s: String = checkNext();
-    return (s.matches(regexp));
-  }
+  private tokenize() {
+    // Remove all newlines
+    let tokenizedProgram = this.program.replace(/\r?\n|\r/g, "");
```

Comment:
Not sure I totally understand this part- if we remove all newlines, will we still be able to report errors with the line number of the code that caused the error?

Reply:
Oh good point.